.github/workflows/maven-publish.yml

```
@@ -19,44 +19,244 @@ jobs:
+          
+          TEMP_DIR=$(mktemp -d)
+          METADATA_FILE="${TEMP_DIR}/maven-metadata.xml"
+          
+          # Download metadata from repository
+          META_URL="${SNAPSHOT_REPO_URL}org/opensearch/${ARTIFACT_ID}/${VERSION}/maven-metadata.xml"
+          echo "Downloading metadata from ${META_URL}"
+          
+          # Try to download the metadata file
+          curl -s -u "${SONATYPE_USERNAME}:${SONATYPE_PASSWORD}" -o "${METADATA_FILE}" "${META_URL}"
+          
+          # If successful, modify and upload back
+          if [ -s "${METADATA_FILE}" ]; then
+            echo "Modifying metadata for ${VERSION}"
+            cp "${METADATA_FILE}" "${METADATA_FILE}.bak"
+          
+            awk -v commit="${COMMIT_ID}" '
+              /<versioning>/ {
+                print $0
+                print "  <commitId>" commit "</commitId>"
+                next
+              }
+              {print}
+            ' "${METADATA_FILE}.bak" > "${METADATA_FILE}"
+          
+            # Upload modified file back
+            curl -X PUT -u "${SONATYPE_USERNAME}:${SONATYPE_PASSWORD}" --upload-file "${METADATA_FILE}" "${META_URL}"
+          
+            # Update the SHA checksums (the publishing script might not have created these)
+            cd "${TEMP_DIR}"
+            sha256sum "maven-metadata.xml" | awk '{print $1}' > "maven-metadata.xml.sha256"
+            sha512sum "maven-metadata.xml" | awk '{print $1}' > "maven-metadata.xml.sha512"
+          
+            # Upload the checksums
+            curl -X PUT -u "${SONATYPE_USERNAME}:${SONATYPE_PASSWORD}" --upload-file "maven-metadata.xml.sha256" "${META_URL}.sha256"
+            curl -X PUT -u "${SONATYPE_USERNAME}:${SONATYPE_PASSWORD}" --upload-file "maven-metadata.xml.sha512" "${META_URL}.sha512"
+            cd -
+          
+            echo "Updated metadata and checksums for ${VERSION}"
+          else
+            echo "Failed to download metadata for ${VERSION}, skipping"
```

shall we fail the CI instead of swallow it?

---

Added code to fail the CI